test/system/interactive_shell.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+#####################
+#! \file interactive_shell.py
+## \verbatim
+## Top contributors (to current version):
+##   Andrew V. Jones
+## This file is part of the CVC4 project.
+## Copyright (c) 2020 by the authors listed in the file AUTHORS
+## in the top-level source directory) and their institutional affiliations.
+## All rights reserved.  See the file COPYING in the top-level source
+## directory for licensing information.\endverbatim
+##
+## \brief A simple test file to interact with CVC4 with line editing
+#####################
+
+import sys
+
+try:
+    import pexpect
+except ImportError:
+    pass
+
+
+def check_iteractive_shell():
+    """
+    Interacts with CVC4's interactive shell and checks that things such a tab
+    completion and "pressing up" works.
+    """
+
+    # Open CVC4
+    child = pexpect.spawnu("bin/cvc4", timeout=1)
+
+    # We expect to see the CVC4 prompt
+    child.expect("CVC4>")
+
+    # If we send a line with just 'BOOLE' ...
+    child.sendline("BOOLE")
+
+    # ... then we get an error
+    child.expect("Parse Error: <shell>:...: Unexpected token: 'BOOLE'")
+
+    # Start sending 'BOOL' (without an E)
+    child.send("BOOL")
+
+    # Send tab twice
+    child.sendcontrol("i")
+    child.sendcontrol("i")
+
+    # We expect to see the completion
+    child.expect("BOOL.*BOOLEAN.*BOOLEXTRACT")
+
+    # NOTE: the double tab has completed our 'BOOL' to 'BOOLE'!
+
+    # Now send enter (which submits 'BOOLE')
+    child.sendcontrol("m")
+
+    # So we expect to see an error for 'BOOLE'
+    child.expect("Parse Error: <shell>:...: Unexpected token: 'BOOLE'")
+
+    # Send enter
+    child.sendcontrol("m")
+
+    # We expect to see the CVC4 prompt
+    child.expect("CVC4>")
+
+    # Now send an up key
+    child.send("\033[A")
+
+    # Send enter
+    child.sendcontrol("m")
+
+    # We expect to see an error on 'BOOLE' again
+    child.expect("Parse Error: <shell>:...: Unexpected token: 'BOOLE'")
+
+    return 0
+
+
+if __name__ == "__main__":
+
+    # By default, we want a zero return code (e.g., if we don't have pexpect,
+    # then we want the test to "pass" by default)
+    ret_val = 0
+
+    # If we have pexpect ...
+    if "pexpect" in sys.modules:
+        # presume the test failed (e.g., due to an exception)
+        ret_val = 1
```

Comment:
I think that if there is an exception, we will not actually proceed to `sys.exit(ret_val)` or am I missing something? If I understood it correctly, then setting `ret_val = 1` here is essentially a noop and could be removed.

Reply:
My original code looked like this:

```
if "pexpect" in sys.modules:
     sys.exit(check_iteractive_shell())

sys.exit(0)
```

shall I just go back to that?

Reply:
Done!